### PR TITLE
Entities + DTOs

### PR DESCRIPTION
**Pull request by EliasJorgensen**
Oprettelse af Entities, Context og medhørende migrations.

Derudover er der også oprettet DTOs.

🥇

### CODE REVIEW

**Comment by EliasJorgensen** [comment]
@oliverjuhl I would argue that they shouldn't all have an Id, because every DTO will not necessarily have an ID. They all have one at the moment, but it is not a requirement for a DTO.

File: PlanningPoker.Entities.Tests/ItemTests.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System.Collections.Generic;
+using Xunit;
+
+namespace PlanningPoker.Entities.Tests
+{
+    public class UnitTest1
```
Comment (VilfredSikker) [inline PlanningPoker.Entities.Tests/ItemTests.cs]: Bad name. Should be self-explanatory

File: PlanningPoker.Entities.Tests/SessionTests.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public class SessionTests
+    {
+        [Fact]
+        public void Items_is_HashSet_of_Item()
+        {
+            var session = new Session();
+
+            var items = session.Items as HashSet<Item>;
+
+            Assert.NotNull(items);
```
Comment (VilfredSikker) [inline PlanningPoker.Entities.Tests/SessionTests.cs]: This isn't really testing that the type is a HashSet, is it? Can we test TypeOf?

File: PlanningPoker.Entities.Tests/SessionTests.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        }
+
+        [Fact]
+        public void Users_is_HashSet_of_User()
+        {
+            var session = new Session();
+
+            var users = session.Users as HashSet<User>;
+
+            Assert.NotNull(users);
```
Comment (VilfredSikker) [inline PlanningPoker.Entities.Tests/SessionTests.cs]: This isn't really testing that the type is a HashSet, is it? Can we test TypeOf?

File: PlanningPoker.Entities/Summary.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+
+namespace PlanningPoker.Entities
+{
+    public class Summary : BaseEntity
+    {
+        public ICollection<ItemEstimate> ItemEstimates { get; set; }
+
+        public int SessionId { get; set; }
```
Comment (VilfredSikker) [inline PlanningPoker.Entities/Summary.cs]: This should be Required as well

File: PlanningPoker.Shared/ItemEstimateDTO.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace PlanningPoker.Shared
+{
+    class ItemEstimateDTO
+    {
+        public int Id { get; set; }
+
+        public string ItemTitle { get; set; }
```
Comment (VilfredSikker) [inline PlanningPoker.Shared/ItemEstimateDTO.cs]: We don't save Description here?

File: PlanningPoker.WebApi.Tests/PlanningPoker.WebApi.Tests.csproj [hunk 6]
```diff
@@ -8,7 +8,6 @@
 
     <ItemGroup>
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.0" />
-        <PackageReference Include="StyleCop.Analyzers" Version="1.0.2" />
```
Comment (VilfredSikker) [inline PlanningPoker.WebApi.Tests/PlanningPoker.WebApi.Tests.csproj]: Tired of yellow lines?

File: PlanningPoker.Shared/ItemEstimateDTO.cs (hunk 5 above)
Comment (EliasJorgensen) [reply PlanningPoker.Shared/ItemEstimateDTO.cs]: According to our google sheet with all the entities etc,  an `ItemEstimate` only contains title and estimate.

**Comment by IMarnild** [comment]
More DTOs are in the making on my branch "Repositories", all Create/Update DTOs will be featured there.

File: PlanningPoker.Entities/Item.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        [Required]
+        public string Title { get; set; }
+
+        public string Description { get; set; }
+
+        public ICollection<Round> Rounds { get; set; }
+
+        public Item()
+        {
+            Rounds = new HashSet<Round>();
```
Comment (Crunchyalex) [inline PlanningPoker.Entities/Item.cs]: Skal der ikke også være en constructor til Item med Title og Desc som parameter? 

Eller er Entities Construction overladt til en anden del?

Maybe I'm overthinking this.

File: PlanningPoker.Entities/Migrations/20181129124117_SessionRefactorMigration.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace PlanningPoker.Entities.Migrations
+{
+    public partial class SessionRefactorMigration : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (Crunchyalex) [inline PlanningPoker.Entities/Migrations/20181129124117_SessionRefactorMigration.cs]: Jeg antager at det her bare skal laves på et andet tidspunkt.

File: PlanningPoker.Entities/Item.cs (hunk 7 above)
Comment (EliasJorgensen) [reply PlanningPoker.Entities/Item.cs]: Jo, men min tanke var at det lavede man bare ad hoc, som man får brug for det.

File: PlanningPoker.Entities/Migrations/20181129124117_SessionRefactorMigration.cs (hunk 8 above)
Comment (EliasJorgensen) [reply PlanningPoker.Entities/Migrations/20181129124117_SessionRefactorMigration.cs]: Alt i migrations mappen er genereret af deres migration tool - ikke noget vi redigerer i. Så dør baby.

**Pull request merged**